packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editing.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { Form } from 'informed';
+
+import { mergeClasses } from '../../../classify';
+import Button from '../../Button';
+import ShippingRadios from './shippingRadios';
+import defaultClasses from './editing.css';
+
+const Editing = props => {
+    const {
+        handleSubmit,
+        hasShippingMethods,
+        isLoading,
+        selectedShippingMethod,
+        shippingMethods
+    } = props;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    return (
+        <div className={classes.root}>
+            <h3 className={classes.heading}>Shipping Method</h3>
+            <Form className={classes.form} onSubmit={handleSubmit}>
+                <ShippingRadios
+                    hasShippingMethods={hasShippingMethods}
+                    isLoading={isLoading}
+                    selectedShippingMethod={selectedShippingMethod}
+                    shippingMethods={shippingMethods}
+                />
+                <Button
+                    priority="normal"
+                    type="submit"
+                    disabled={isLoading || !hasShippingMethods}
```

Comment:
This should also be disabled if the checkout page `isUpdating` and while the submit mutation is in flight, otherwise I can spam the button and it'll queue a bunch of mutations!

[![Image from Gyazo](https://i.gyazo.com/b411aa15e1f6d53710bfcb3f6ff7b1fb.gif)](https://gyazo.com/b411aa15e1f6d53710bfcb3f6ff7b1fb)

Reply:
Fixed in https://github.com/magento/pwa-studio/pull/2280/commits/963ed01710348091380217d1054bd64d69613e82.